website/source/docs/vault-enterprise/hsm/behavior.html.md

```diff
@@ -0,0 +1,88 @@
+---
+layout: "docs"
+page_title: "Vault Enterprise HSM Behavioral Changes"
+sidebar_current: "docs-vault-enterprise-hsm-behavior"
+description: |-
+  Vault Enterprise HSM support changes the way Vault works with regard to unseal and recovery keys as well as rekey and recovery operations.
+---
+
+# Vault Enterprise HSM Behavioral Changes
+
+This page contains information about the behavioral differences that take
+effect when using Vault with an HSM.
+
+## Key Split Between Unseal Keys and Recovery Keys
+
+Normally, Vault uses a single set of unseal keys to perform both decryption of
+the cryptographic barrier and to authorize recovery operations, such as the
+[`generate-root`](/docs/http/sys-generate-root.html)
+functionality.
+
+When using an HSM, because the HSM automatically unseals the barrier but
+recovery operations should still have human oversight, Vault instead uses two
+sets of keys: unseal keys and recovery keys.
+
+## Unseal (Master) Key
+
+Vault usually generates a master key and splits it using [Shamir's Secret
+Sharing](https://en.wikipedia.org/wiki/Shamir%27s_Secret_Sharing) to prevent a
+single operator from being able to modify and unseal Vault (see more
+information about Vault's security model
+[here](/docs/internals/security.html)).
+
+When using an HSM, Vault instead stores the master key, encrypted by the HSM,
+into its internal storage. As a result, during an `init` command, the number of
+key shares, threshold, and stored shares are required to be set to `1`, meaning
+to not split the master key, so that the single key share is itself the master
+key. (Vault does not do this automatically as it generally prefers to error
+rather than change parameters set by an operator.)
+
+Vault does not currently support rekeying the master key when protected by an
```

Comment:
"It's tricky". When the primary is protected by an HSM we do not support rekeys; however, when a secondary is HSM protected we do, because basically the secondary will do the right thing when a rekey happens.